ambari-server/src/main/java/org/apache/ambari/server/state/ServiceInfo.java

```diff
@@ -469,6 +471,56 @@ public void setProperties(List<PropertyInfo> properties) {
     this.properties = properties;
   }
 
+  @Override
+  public Object clone() throws CloneNotSupportedException {
+    ServiceInfo clone = (ServiceInfo) super.clone();
```

Comment:
I don't see where you're cloning and need this method.

Reply:
@ncole I clone the service info in StackMerger tool to create the flattened ServiceInfo.